test/MonthSelectTest.php

```diff
@@ -14,8 +13,8 @@ class MonthSelectTest extends TestCase
     #[DataProvider('provideFilter')]
     public function testFilter(array $options, array $input, ?string $expected): void
     {
-        $sut = new MonthSelectFilter();
-        $sut->setOptions($options);
+        /** @psalm-suppress MixedArgumentTypeCoercion */
```

Comment:
If you `@psalm-import-type Options from MonthSelect` on the class, add `@param Options $options` here and `@return list<array{0: Options, 1: mixed, 2: string|null}>` on the data provider, no inline suppressions will be needed. As a general rule, only ever add inline supressions to code when you are knowingly providing an invalid argument to force an exception or something in a test. Issues belong in the baseline when they are not getting fixed 👍